python/pyarrow/dataset.py

```diff
@@ -713,12 +735,15 @@ def write_dataset(data, base_dir, basename_template=None, format=None,
     elif isinstance(data, (pa.Table, pa.RecordBatch)):
         schema = schema or data.schema
         data = [data]
-    elif isinstance(data, list):
+    elif isinstance(data, (list, tuple)):
         schema = schema or data[0].schema
```

Comment:
Could also pass this to an InMemoryDataset? Then `_filesystemdataset_write` could potentially be simplified to not have to deal with a list of batches/tables